### PR TITLE
Header

### PR DESCRIPTION
**Pull request by EllieLilithFox**
convert header to functional component and add tests

### CODE REVIEW

File: gearcatalog/frontend/src/actions/actions.types.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import { Gear, GearNoId } from '../components/gear/types';
 
 // TODO: #9 Create correct type for User
-export type User = Record<string, unknown>;
+export type User = {
+  username: string,
```
Comment (NatalieWolfe) [inline gearcatalog/frontend/src/actions/actions.types.ts:5]: Users don't have IDs?

File: gearcatalog/frontend/src/components/layout/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  user: User | null,
+};
+
+const Header = ({ logout, isAuthenticated, token, user }: HeaderProps) => {
+  const authLinks = (
+    <ul className="navbar-nav ml-auto mt-2 mt-lg-0">
+      <span className="navbar-text text-light mr-3">
+        <strong>
+          {user ? `Welcome ${user.username[0].toUpperCase() +
+          user.username.slice(1)}` : ""}
```
Comment (NatalieWolfe) [inline gearcatalog/frontend/src/components/layout/Header.tsx:21]: What if I want my username to be all lowercase!?

Comment (EllieLilithFox) [reply gearcatalog/frontend/src/components/layout/Header.tsx:21]: \>.>;;

File: gearcatalog/frontend/src/actions/actions.types.ts (hunk 1 above)
Comment (EllieLilithFox) [reply gearcatalog/frontend/src/actions/actions.types.ts:5]: They do, I just don't use them on the frontend as of right now. I figured I would just type what I'm currently using.

**Pull request merged**